pkg/api/types.go

```diff
@@ -1375,8 +1375,42 @@ type BuildArg struct {
 	// Name of the build arg.
 	Name string `json:"name,omitempty"`
 
-	// Value of the build arg.
+	// Value of the build arg. Cannot be set if ValueFrom is set.
 	Value string `json:"value,omitempty"`
+
+	// ValueFrom specifies the source of the build Arg. Cannot be set if Value is set.
+	ValueFrom *BuildArgSource `json:"value_from,omitempty"`
+}
+
+// BuildArgSource represents a source for the value of a BuildArg.
+type BuildArgSource struct {
+	// SecretKeyRef selects a key of a secret
+	SecretKeyRef *SecretKeySelector `json:"secret_key_ref,omitempty"`
+	// Namespace of the BuildArg source
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Also why are these `omitempty` when we aggressively validate that they are not empty?

Reply:
Removed.